ui/components/src/ui-components/table/component/Table.tsx

```diff
@@ -909,12 +909,10 @@ export class Table extends React.Component<TableProps, TableState> {
         this.props.onRowsLoaded(index, index + loadResult.rows.length - 1);
 
       const showFilter = this._isShowFilterRow();
```

Comment:
The correct code is:

      const showFilter = this._isShowFilterRow();
      if (showFilter !== this._filterRowShown)
        this.toggleFilterRow(showFilter);
      if (showFilter)
        await this.loadDistinctValues();